cpp/src/arrow/parquet/reader.cc

```diff
@@ -103,6 +108,23 @@ Status FileReader::Impl::ReadFlatColumn(int i, std::shared_ptr<Array>* out) {
   return flat_column_reader->NextBatch(reader_->num_rows(), out);
 }
 
+Status FileReader::Impl::ReadFlatTable(std::shared_ptr<Table>* table) {
+  const std::string name = reader_->descr()->schema()->name();
```

Comment:
i'm not familiar with this method chain, but is the copy necessary here (i.e. does name return a reference or a string?